app/models/smart_proxy.rb

```diff
@@ -88,12 +86,18 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
-  private
-
   def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+    @proxy_status ||= { :version => ProxyStatus::Version.new(self) }
+    features.map(&:name).each do |feature|
+      normalized_feature_name = feature.delete(' ')
+      status = ProxyStatus.find_status_by_humanized_name(normalized_feature_name)
+      @proxy_status[normalized_feature_name.downcase.to_sym] = status.new(self) if status
```

Comment:
also maybe consider updating the feature validation? so only one feature with the same name (case insensitive ) can be created?